builtin/providers/aws/resource_aws_elasticache_cluster.go

```diff
@@ -336,6 +337,13 @@ func CacheClusterStateRefreshFunc(conn *elasticache.ElastiCache, clusterID, give
 
 		// return given state if it's not in pending
 		if givenState != "" {
+			// loop the nodes and check their status as well
+			for _, n := range c.CacheNodes {
```

Comment:
I suppose in theory that could happen. In my extensive testing of multiple regions, node counts, and types (redis/memcached), I've not seen this happen. But I'll add code to check that as well